controllers/reconcile.go

```diff
@@ -132,6 +132,146 @@ func populateContainer(d *apps.Deployment, frontend *crd.Frontend, frontendEnvir
 	}
 }
 
+// getAkamaiSecret gets the akamai secret from the cluster
+func getAkamaiSecret(ctx context.Context, client client.Client, frontend *crd.Frontend) (*v1.Secret, error) {
+	secret := &v1.Secret{}
+	err := client.Get(ctx, types.NamespacedName{Name: "akamai", Namespace: frontend.Namespace}, secret)
+	if err != nil {
+		return nil, err
+	}
+	return secret, nil
+}
+
+// constructAkamaiEdgercFileFromSecret constructs the akamai edgerc file from the secret
+func makeAkamaiEdgercFileFromSecret(secret *v1.Secret) string {
+	edgercFile := "[default]\n"
+	edgercFile += fmt.Sprintf("host = %s\n", secret.Data["host"])
+	edgercFile += fmt.Sprintf("access_token = %s\n", secret.Data["access_token"])
+	edgercFile += fmt.Sprintf("client_token = %s\n", secret.Data["client_token"])
+	edgercFile += fmt.Sprintf("client_secret = %s\n", secret.Data["client_secret"])
+	edgercFile += "[ccu]\n"
+	edgercFile += fmt.Sprintf("host = %s\n", secret.Data["host"])
+	edgercFile += fmt.Sprintf("access_token = %s\n", secret.Data["access_token"])
+	edgercFile += fmt.Sprintf("client_token = %s\n", secret.Data["client_token"])
+	edgercFile += fmt.Sprintf("client_secret = %s\n", secret.Data["client_secret"])
+	return edgercFile
+}
+
+func createCachePurgePathList(frontend *crd.Frontend, frontendEnvironment *crd.FrontendEnvironment) []string {
+	var purgeHost string
+	// If the cache bust URL doesn't begin with https:// then add it
+	if strings.HasPrefix(frontendEnvironment.Spec.AkamaiCacheBustURL, "https://") {
+		purgeHost = frontendEnvironment.Spec.AkamaiCacheBustURL
+	} else {
+		purgeHost = fmt.Sprintf("https://%s", frontendEnvironment.Spec.AkamaiCacheBustURL)
+	}
+
+	// If purgeHost ends with a / then remove it
+	purgeHost = strings.TrimSuffix(purgeHost, "/")
+
+	frontendName := frontend.Name
+	purgePaths := []string{}
+
+	// If there is no purge list return the default
+	if frontend.Spec.AkamaiCacheBustPaths == nil {
+		defaultPurgePath := fmt.Sprintf("%s/apps/%s/fed-mods.json", purgeHost, frontendName)
```

Comment:
Did we speak before about ALWAYS having this in the list, in which case, the default purgePaths `[]string{}` could be prepopulated and we could remove this block entirely?